public/app/plugins/panel/timeseries/utils.ts

```diff
@@ -17,6 +17,41 @@ import { applyNullInsertThreshold } from '@grafana/ui/src/components/GraphNG/nul
 import { nullToValue } from '@grafana/ui/src/components/GraphNG/nullToValue';
 import { partitionByValuesTransformer } from 'app/features/transformers/partitionByValues/partitionByValues';
 
+// mutates fields!
+function unifyEnumFields(frames: DataFrame[]) {
+  let allTexts: string[] = [];
+
+  let frames2: DataFrame[] = frames.map((frame) => {
+    return {
+      ...frame,
+      fields: frame.fields.map((field) => {
+        if (field.type === FieldType.enum) {
```

Comment:
maybe we should check that any enum field exists first?  then we can avoid all the copies for the 99% chance they do not 🤔

Reply:
i dont think there's a lot of perf to gain here. it's basically just `frame.fields.map((field) => field)` for what, like < 30 items?

Reply:
maybe... but I am actually talking about the step above it that will be: `frame => { ...frame }` for as many series as prometheus returns 🤷 30?

Reply:
i'm not fundamentally opposed to adding a precheck :)